docs/reference/modules/discovery.asciidoc

```diff
@@ -1,30 +1,765 @@
 [[modules-discovery]]
-== Discovery
+== Discovery and cluster formation
```

Comment:
May we also update the glossary? 
It would be useful to see definition of terms like "master-eligible", "master", "quorum", "follower", "leader". For example, it's not immediately clear if a follower is any non-master node in the cluster or only non-master master-eligible node.
https://www.elastic.co/guide/en/elasticsearch/reference/current/glossary.html

Reply:
I pushed 02b7ebd86e020e4caf1c7c3fc1abfe00c5802efd to reduce the usage of the leader/follower terminology, and this also defines the terms "leader check" and "follower check".

The terms "master" and "master-eligible" are [defined here](https://github.com/elastic/elasticsearch/blob/c556b9ef191d660c6b4e820cdc427481eecd3eb4/docs/reference/modules/node.asciidoc#node).

The term "quorum" is reasonably standard, and [defined where it first appears](https://github.com/DaveCTurner/elasticsearch/blob/02b7ebd86e020e4caf1c7c3fc1abfe00c5802efd/docs/reference/modules/discovery.asciidoc#quorum-based-decision-making) (ignoring occurrences in headings).